specs/Workers.md

```diff
@@ -0,0 +1,1228 @@
+  /// By default, the scope value for a service worker registration is set to the directory where the service worker script is located.
+  ///
+  /// The `scope` parameter will be a fully qualified URL that includes both the origin (scheme, host, and optionally port) and the path.
+  /// If the origin and scope includes an internationalized domain name (IDN), it should be represented in punycode format.
+  ///
+  /// For example, if the service worker script is located at https://example.com/sw.js, the default scope is https://example.com/, and the service worker controls all pages under https://example.com/
+  /// If the scope is set to /app/ when registering the service worker, the scope will be https://example.com/app/.
+  ///
+  /// This corresponds to the `scope` property of the `ServiceWorkerRegistration` object in the DOM.
+  /// For more information, see the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/scope).
+  ///
+  /// The caller must free the returned string with `CoTaskMemFree`.  See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  [propget] HRESULT Scope([out, retval] LPWSTR* value);
+
+  /// The service worker that was registered.
+  HRESULT GetServiceWorkerRegistration(
+      [out, retval] ICoreWebView2ServiceWorkerRegistration** value
+  );
+}
+
+/// A collection of ICoreWebView2ServiceWorkerRegistration.
+[uuid(9860773f-16e5-5521-9bc2-a6d8e5347d4a), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegistrationCollectionView : IUnknown {
+  /// The number of elements contained in the collection.
+  [propget] HRESULT Count([out, retval] UINT32* value);
+
+  /// Gets the element at the given index.
+  HRESULT GetValueAtIndex([in] UINT32 index, [out, retval] ICoreWebView2ServiceWorkerRegistration** value);
+}
+
+/// Receives the result of the `GetServiceWorker` method.
+[uuid(bd120b47-b859-5f44-9913-0318130ffb69), object, pointer_default(unique)]
+interface ICoreWebView2GetServiceWorkerCompletedHandler : IUnknown {
+
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] ICoreWebView2ServiceWorker* result);
+}
+
+/// Receives `Destroyed` events.
+[uuid(9d79117e-4e96-5a53-9933-d7a31d0106ae), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegistrationDestroyedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.
+  HRESULT Invoke(
+      [in] ICoreWebView2ServiceWorkerRegistration* sender,
+      [in] IUnknown* args);
+}
+
+[uuid(0c0b03bd-ced2-5518-851a-3d3f71fb5c4b), object, pointer_default(unique)]
+interface ICoreWebView2ServiceWorkerRegistration : IUnknown {
+  /// A string representing a URL that defines a service worker's registration scope. It is relative to the base URL of the application.
```

Comment:
This text says the property is relative. The next paragraph says it is a fully qualified URL. The fourth paragraph says it corresponds to the ServiceWorkerRegistration's scope which at least when calling `register` the scope is relative. Please make this clearer. This also applies to the above definition of scope in the registered event args.

Reply:
Updated, during registration the scope is relative but in our API's it is full URL.